research/context/preselection/preselect_wikipedia_comments.py

```diff
@@ -0,0 +1,125 @@
+# coding=utf-8
+# Copyright 2018 The Conversation-AI.github.io Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Preselect Wikipedia comments.
+
+A library to preselect wikipedia comments based on their toxicity score
+among others aspects.
+
+The initial goal was to create a sample of comments to use for annotation;
+the sample was stratified to ensure existence of toxic comments. If one
+uses 3 zones, then [0,1] will be divided in [0,0.33), [0.33,0.66) and
+[0.66,1], so that the comments are scored from all three zones.
+
+The BigQuery table queried to create this sample contains 240M rows (294Gb)
+and we assume that 'toxicity', 'cleaned_content', 'replyTo_id' and 'type'
+attributes exist.
+
+Run as:
+python preselect_wikipedia_comments.py \
+--gbq_project_name "wikidetox-viz" \
+--gbq_table_name "wikiconv_v2.en_20180701_external" \
+--output_csv_path "sample.csv" \
+--limit 100 \
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tqdm import tqdm
+import pandas as pd
+import tensorflow as tf
+from google.cloud import bigquery
+
+FLAGS = tf.app.flags.FLAGS
+tf.app.flags.DEFINE_integer("limit", 100, "Number of comments per zone (zones_num*limit + 1).")
```

Comment:
I'm confused by what the `zones_num*limit + 1` is intended to tell me; it's obviously not a definition for "limit" as it would be self referential in a unbounded way.